app/views/ChooseProvider.js

```diff
@@ -367,16 +368,19 @@ class ChooseProviderScreen extends Component {
                   let key = this.state.authoritiesList.indexOf(item);
 
                   return (
-                    <MenuOption
-                      key={key}
-                      onSelect={() => {
-                        this.addAuthorityToState(name);
-                      }}
-                      disabled={this.state.authoritiesList.length === 1}>
-                      <Typography style={styles.menuOptionText} use={'body2'}>
-                        {name}
-                      </Typography>
-                    </MenuOption>
+                    _.findIndex(this.state.selectedAuthorities, ['key', name]) <
```

Comment:
So - this whole section was pretty messy before you arrived - but do you think you could pull it out into a new component, something like `<AuthoritiesList />`?  Basically lines 348 - 397.

I know that it's a little out-of-scope with the original ticket, but it feels like a good opportunity to get this cleaned up.

Reply:
Also - could you pull the `_.findIndex()` method into a separate method, something like `isAuthoritySelected()`? It's a little hard to understand what that code is doing at a quick glance.

Reply:
> Also - could you pull the `_.findIndex()` method into a separate method, something like `isAuthoritySelected()`? It's a little hard to understand what that code is doing at a quick glance.
Okey, I'll update this component.